platforms/linux/arm.toolchain.cmake

```diff
@@ -57,6 +57,16 @@ if(NOT DEFINED CMAKE_CXX_FLAGS)
 
   set(CMAKE_CXX_FLAGS           "${CMAKE_CXX_FLAGS} -fdata-sections -Wa,--noexecstack -fsigned-char -Wno-psabi")
   set(CMAKE_C_FLAGS             "${CMAKE_C_FLAGS} -fdata-sections -Wa,--noexecstack -fsigned-char -Wno-psabi")
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    # see https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html
+    if(ENABLE_BF16)
```

Comment:
> ENABLE_BF16

Toolchain files have issue with cached variables (not available to them - just add `message()`),

Reply:
I've checked this variable, it seem to be working. I build with `-DENABLE_BF16=ON` and got `-march=armv8.6-a` in the compiler string. We use this approach in XuanTie cmake toolchain: https://github.com/opencv/opencv/blob/4.x/platforms/linux/riscv64-071-gcc.toolchain.cmake (`CORE` variable)

Reply:
Just add `message()` and run.
Also add `message() for modified `CMAKE_CXX_FLAGS` because you got the `else()` branch in case of undefined value.

Reply:
Oh, I remembered, it was an issue with CMAKE_CXX_FLAGS. As I understand, modern cmake (3.7+) have respective `_INIT` variables for toolchain files specifically (https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_INIT.html). In new commit I've updated some of our toolchains to use them too.